prover/proof_submitter/transaction/builder.go

```diff
@@ -73,8 +78,13 @@ func (a *ProveBlockTxBuilder) Build(
 				return nil, ErrUnretryableSubmission
 			}
 		} else {
-			to = a.guardianProverAddress
-
+			if tier > encoding.TierGuardianMinorityID {
+				to = a.majorityGuardianProverAddress
+			} else if tier == encoding.TierGuardianMinorityID && a.minorityGuardianProverAddress != ZeroAddress {
+				to = a.minorityGuardianProverAddress
+			} else {
+				return nil, fmt.Errorf("tier %d need set minorityGuardianProverAddress", tier)
```

Comment:
If `tier < encoding.TierGuardianMinorityID` and `a.majorityGuardianProverAddress ~= common.Address{}` then will return error message, but this error content is not correct.

Reply:
The switch case only happens in `guardian` status and `guardian = tier >= encoding.TierGuardianMinorityID`.